consensus/src/member.rs

```diff
@@ -327,22 +331,25 @@ where
                 TaskDetails::Cancel => (),
                 TaskDetails::Perform {
                     message,
-                    recipient,
+                    recipients,
                     reschedule,
                 } => {
-                    self.send_unit_message(message, recipient);
+                    for recipient in recipients.into_iter() {
+                        self.send_unit_message(message.clone(), recipient);
+                    }
+
                     task.counter += 1;
                     self.task_queue.schedule_in(task, reschedule)
                 }
             }
         }
     }
 
-    fn random_peer(&self) -> Recipient {
-        let node = rand::thread_rng()
-            .gen_range(0..self.n_members.into())
-            .into();
-        Recipient::Node(node)
+    fn random_peers(&self, n: usize) -> Vec<Recipient> {
+        self.peers
+            .iter()
+            .cloned()
+            .choose_multiple(&mut rand::thread_rng(), n)
```

Comment:
This uses `choose_multiple` on iterators and not slices and thus has `O(self.peers.len())` complexity :P Iterators don't give random access to individual indices, hence it cannot be possibly better.  I guess you need to shuffle the ordering of the calls here to make it right.

Reply:
:+1: